### PR TITLE
Change string/concatenate to support mulitple inputs

### PR DESCRIPTION
**Pull request by otaviobp**
String concatenate was only concatenating 2 strings. Now we are able to
concatenate up to 32 strings using a single node. The approach taken is
the same approach from int opeartions that have multiple inputs

Signed-off-by: Otavio Pontes <otavio.pontes@intel.com>

### CODE REVIEW

File: src/modules/flow/string/string-ascii.c [hunk 1]
```diff
@@ -146,27 +181,45 @@ string_concat(struct sol_flow_node *node,
...
 
-    dest = strcpy(dest, mdata->base.string[0]);
+    for (i = 0; i < 32; i++) {
+        if (!(mdata->var_initialized & (1u << i)))
+            continue;
 
-    if (mdata->separator)
-        dest = strcat(dest, mdata->separator);
+        if (count && mdata->separator) {
+            dest = strncat(dest, mdata->separator, len);
```
Comment (lpereira) [inline src/modules/flow/string/string-ascii.c]: Why not use `sol_buffer`?

Comment (otaviobp) [inline src/modules/flow/string/string-ascii.c]: I was planning on using buffer, but then I just followed the approach that was being used before. But I think that it really better use buffer for that.

**Comment by otaviobp** [comment]
lol. the idea in the title of the commit message is the oposite. Correct:

strings: Rename bytes option to chars in concatenate and compare

Unicode strings are counting the number of characters and not the number
of bytes.

**Comment by glima** [comment]
``FAILED running string-concat.fbp: wrong exit code, got 1 but expected 0
Output:
==14433== WARNING: AddressSanitizer failed to allocate 0xc00800018078 bytes``

check the bot's output. Enable the sanitizers on your machine.

**Comment by bdilly** [comment]
@otaviobp wasn't the conclusion of our discussion on IRC yesterday with @glima to completely remove this counting option?

So we would get rid of this logic inside this node. It's specially true imo if you're going to fix the size only of the final string. It can be easily achieved using string/slice later, right? Something like:

concat(string/concat) OUT -> IN _(string/slice:end=CHARS) OUT -> IN _(console)

instead of

concat(string/concat:chars=CHARS) OUT -> IN _(console)

@glima, what do you think about it?

**Comment by otaviobp** [comment]
For me it is ok to remove it. I had understood that I was suppose to keep it, but limiting the size of the full string.

**Comment by glima** [comment]
I was ok with the semantics changed to the full size, but removing it altogether is cool as well. I vote for removing code :)

**Pull request closed without merging**